yt_dlp/extractor/ant1newsgr.py

```diff
@@ -0,0 +1,184 @@
+    _VALID_URL = r'https?://(?:www\.)?ant1news\.gr/[^/]+/article/(?P<id>\d+)/'
+
+    _TESTS = [{
+        'url': 'https://www.ant1news.gr/afieromata/article/549468/o-tzeims-mpont-sta-meteora-oi-apeiles-kai-o-xesikomos-ton-kalogeron',
+        'md5': '294f18331bb516539d72d85a82887dcc',
+        'info_dict': {
+            'id': '_xvg/m_cmbatw=',
+            'ext': 'mp4',
+            'title': 'md5:a93e8ecf2e4073bfdffcb38f59945411',
+            'timestamp': 1603092840,
+            'upload_date': '20201019',
+            'thumbnail': 'https://ant1media.azureedge.net/imgHandler/1920/756206d2-d640-40e2-b201-3555abdfc0db.jpg',
+        },
+    }, {
+        'url': 'https://ant1news.gr/Society/article/620286/symmoria-anilikon-dikigoros-thymaton-ithelan-na-toys-apoteleiosoyn',
+        'info_dict': {
+            'id': '620286',
+            'title': 'md5:91fe569e952e4d146485740ae927662b',
+        },
+        'playlist_mincount': 2,
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        info = self._search_json_ld(webpage, video_id, expected_type='NewsArticle')
+        embed_urls = list(Ant1NewsGrEmbedIE._extract_urls(webpage, url, **info))
+        if not embed_urls:
+            raise ExtractorError('no videos found for %s' % video_id)
+        if len(embed_urls) == 1:
+            return self.url_result(embed_urls[0], ie=Ant1NewsGrEmbedIE.ie_key(),
+                                   video_title=info['title'])
+        return self.playlist_from_matches(
+            embed_urls, video_id, info['title'], ie=Ant1NewsGrEmbedIE.ie_key())
+
+
+class Ant1NewsGrEmbedIE(Ant1NewsGrBaseIE):
+    IE_NAME = 'ant1newsgr:embed'
+    IE_DESC = 'ant1news.gr embedded videos'
+    _VALID_URL = r'''(?x)https?://(?:[a-zA-Z0-9\-]+\.)?
+        (?:antenna|ant1news)\.gr/templates/pages/player
+        \?(?:(?:cid=(?P<id>[^&#]+)|[^&=#]+=[^&#]+)&?)+'''
+    _API_PATH = '/news/templates/data/jsonPlayer'
+
+    _TEST = {
+        'url': 'https://www.antenna.gr/templates/pages/player?cid=3f_li_c_az_jw_y_u=&w=670&h=377',
+        'md5': '12872b12af18b5dbf76528786728de8c',
+        'info_dict': {
+            'id': '3f_li_c_az_jw_y_u=',
+            'ext': 'mp4',
+            'title': 'md5:a30c93332455f53e1e84ae0724f0adf7',
+        },
+    }
+
+    @classmethod
+    def _extract_urls(cls, webpage, origin_url=None, **parent_info):
+        # make the scheme in _VALID_URL optional
+        _URL_RE = r'(?:https?:)?//' + cls._VALID_URL.split('://', 1)[1]
+        # simplify the query string part of _VALID_URL; after extracting iframe
+        # src, the URL will be matched again
+        _URL_RE = _URL_RE.split(r'\?', 1)[0] + r'\?(?:(?!(?P=_q1)).)+'
+        EMBED_RE = r'''(?x)
+            <iframe[^>]+?src=(?P<_q1>%(quot_re)s)(?P<url>%(url_re)s)(?P=_q1)
+        ''' % {'quot_re': r'["\']', 'url_re': _URL_RE}
```

Comment:
extremely similar to the other PR. maybe this should be generalized into a common function. I'm sure I have seen similar patterns in other extractors too. 

Just noting down, not actually in scope of this PR

Reply:
I will nevertheless inline `quot_re` as suggested elsewhere.

Reply:
After giving up on scheme matching, I think I have now simplified `EMBED_RE` as much as possible.

Reply:
After rebasing and fixing merge conflicts (c7b91ddea0781abf1cc4a9550e9dd4b0713e57fe was replaced by 311b6615d85d3530f2709c50e4223ff3b6b14361) I realized I still need a way to call `playlist_from_matches` and have it set `url_transparent` and override some fields. So I added `video_kwargs` to it and also added `url_result_or_playlist_from_matches`, a wrapper that either calls `url_result` or the former method.